tests/test_filesystem.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import unicode_literals, division, absolute_import
+from tests import FlexGetBase
+
+
+class TestFilesystem(FlexGetBase):
+    # TODO test symlnks
+    base = "filesystem_test_dir/"
+
+    __yaml__ = """
+        tasks:
+          string:
+            filesystem: """ + base + """Test1
+
+          list:
+           filesystem:
+             - """ + base + """Test1
+             - """ + base + """Test2
+
+          object_string:
+            filesystem:
+              path: """ + base + """Test1
+
+          object_list:
+            filesystem:
+              path:
+                - """ + base + """Test1
+                - """ + base + """Test2
+
+          file_mask:
+            filesystem:
+              path: """ + base + """Test1
+              mask: '*.mkv'
+
+          regexp_test:
+            filesystem:
+              path: """ + base + """Test1
+              regexp: '.*\.(mkv)$'
+
+          recursive_true:
+            filesystem:
+              path: """ + base + """Test1
+              recursive: yes
+
+          recursive_2_levels:
+            filesystem:
+              path: """ + base + """Test1
+              recursive: 2
+
+          retrieve_files:
+            filesystem:
+              path: """ + base + """Test1
+              retrieve: files
+
+          retrieve_files_and_dirs:
+            filesystem:
+              path: """ + base + """Test1
+              retrieve:
+                - files
+                - dirs
+
+          combine_1:
+            filesystem:
+              path: """ + base + """Test1
+              mask: '*.mkv'
+              recursive: 2
+
+          combine_2:
+            filesystem:
+              path: """ + base + """Test1
+              recursive: yes
+              retrieve: dirs
+        """
+
+    item_list = ['file1.mkv', 'file2.txt', 'file10.mkv', 'file11.txt', 'file4.avi', 'file3.xlsx', 'file5.mkv', 'dir1',
+                 'dir2', 'dir3', 'dir4', 'dir6', 'dir7', 'dir8']
+
+    def assert_check(self, task_name, test_type, filenames):
+        self.execute_task(task_name)
+        for file in filenames:
+            if test_type == 'positive':
+                assertion_error = 'Failed {} test, did not find {}'.format(task_name, file)
+                assert self.task.find_entry(filename=file), assertion_error
+            else:
+                assertion_error = 'Failed {} test, found {}'.format(task_name, file)
+                assert not self.task.find_entry(filename=file), assertion_error
+
+    def test_string(self):
+        task_name = 'string'
+        should_exist = 'dir1', 'dir2', 'dir3', 'file1.mkv', 'file2.txt'
+        should_not_exist = [item for item in self.item_list if item not in should_exist]
+
+        self.assert_check(task_name, 'positive', should_exist)
+        self.assert_check(task_name, 'negative', should_not_exist)
```

Comment:
This is going to execute the task twice, I don't think that's what we want. I think the whole `assert_check` makes these tests quite a bit less readable too.